stellargraph/mapper/node_mappers.py

```diff
@@ -1128,3 +1128,275 @@ def flow_from_dataframe(self, node_targets, shuffle=False):
+    on relational graphs that require full-batch training (e.g., RGCN).
+    Use this class with the Keras methods :meth:`keras.Model.fit_generator`,
+        :meth:`keras.Model.evaluate_generator`, and
+        :meth:`keras.Model.predict_generator`,
+
+    This class uses sparse representations to send data to the models.
+
+    This class should be created using the `.flow(...)` method of
+    :class:`RelationalFullBatchNodeGenerator`.
+
+    Args:
+        features (np.ndarray): An array of node features of size (N x F),
+            where N is the number of nodes in the graph, F is the node feature size
+        As (list of sparse matrices): A list of length R of adjacency matrices of the graph of size (N x N)
+            where R is the number of relationships in the graph.
+        targets (np.ndarray, optional): An optional array of node targets of size (N x C),
+            where C is the target size (e.g., number of classes for one-hot class targets)
+        indices (np.ndarray, optional): Array of indices to the feature and adjacency matrix
+            of the targets. Required if targets is not None.
+    """
+
+    use_sparse = True
+
+    def __init__(self, features, As, targets=None, indices=None):
+
+        if (targets is not None) and (len(indices) != len(targets)):
+            raise ValueError(
+                "When passed together targets and indices should be the same length."
+            )
+
+        # Store features and targets as np.ndarray
+        self.features = np.asanyarray(features)
+        self.target_indices = np.asanyarray(indices)
+
+        # convert the list of adjacency matrices to a list of index array
+        # and a list of value array - add batch dimension 1 to each index and value array
+        self.A_indices = [
+            np.expand_dims(np.hstack((A.row[:, None], A.col[:, None])), 0) for A in As
+        ]
+        self.A_values = [np.expand_dims(A.data, 0) for A in As]
+
+        # Reshape all inputs to have batch dimension of 1
+        self.target_indices = np.reshape(
+            self.target_indices, (1,) + self.target_indices.shape
+        )
+
+        self.features = np.reshape(self.features, (1,) + self.features.shape)
+        self.inputs = (
+            [self.features, self.target_indices] + self.A_indices + self.A_values
+        )
+
+        # Reshape all inputs to have batch dimension of 1
+        self.target_indices = np.reshape(
+            self.target_indices, (1,) + self.target_indices.shape
+        )
+        self.features = np.reshape(self.features, (1,) + self.features.shape)
+
+        if targets is not None:
+            self.targets = np.asanyarray(targets)
+            self.targets = np.reshape(self.targets, (1,) + self.targets.shape)
+        else:
+            self.targets = None
+
+    def __len__(self):
+        return 1
+
+    def __getitem__(self, index):
+        return self.inputs, self.targets
+
+
+class RelationalDenseFullBatchNodeSequence(Sequence):
```

Comment:
This class looks nearly identical to the Sparse version above with 3-4 lines different to handle the different types of adjacency matrices.

Is there are a reason why we cannot combine them into a single class with a `sparse` parameter with default value `True`?

Reply:
Great idea, I was copying the API for the `FullBatchNodeSequence` but this sounds much better. I'll implement this.